configurable db connection parameters

nopcoder opened the pull request:
Closes #443

arielshaqed commented on catalog/cataloger_retention_test.go:
```diff
@@ -653,7 +655,7 @@ func TestCataloger_MarkObjectsForDeletion(t *testing.T) {
 	}
 
-	conn, err := db.ConnectDB("pgx", c.DbConnURI)
+	conn, err := db.ConnectDB(params.Database{URI: c.DbConnURI})
```
`"pgx"`?

nopcoder replied:
made it the default on connect - I'm thinking if there is a reason why we would ever want to configure it at this level.
what do you think?

arielshaqed replied:
I'd expect us to configure it to whatever is useful.  I mean, AFAICT the two main Golang Postgres drivers are pg, pq and pgx...  I don't understand the details well enough though.  If nobody will ever use a driver other than pgx (for instance if the other two won't work for what we need) then obviously an unchangeable value is good.
E.g. if you can select the driver (rather than just the protocol) from the URL then it's obviously good enough.  BTW, this applies to _all_ connections not just this one :-/

arielshaqed commented on config/config.go:
```diff
@@ -107,7 +107,12 @@ func setDefaults() {
 func (c *Config) GetDatabaseParams() dbparams.Database {
-	return dbparams.Database{DatabaseURI: viper.GetString("database.connection_string")}
+	return dbparams.Database{
+		URI:                   viper.GetString("database.connection_string"),
```
Postgres connection strings can be very non-URI-ish.

nopcoder replied:
I'll rename it to ConnectionString to make it clear

arielshaqed reviewed (approved):
1. Not sure why we dropped `"pgx"` throughout.
2. UX/Docs/param names: Need to decide whether we ask the user to pass a URI or a connection string, which is [more general](https://www.postgresql.org/docs/12/libpq-connect.html#LIBPQ-CONNSTRING).

arielshaqed commented:
Also (of course) thanks!

nopcoder commented:
> 1. Not sure why we dropped `"pgx"` throughout.
> 2. UX/Docs/param names: Need to decide whether we ask the user to pass a URI or a connection string, which is [more general](https://www.postgresql.org/docs/12/libpq-connect.html#LIBPQ-CONNSTRING).

1. Not dropped, use "pgx" by default if not specified otherwise.  Note that we will have to compile another package in order to configure it at runtime.
2. updated the code to "ConnectionString"

arielshaqed commented on api/handler_test.go:
```diff
@@ -86,7 +86,7 @@ func getHandler(t *testing.T, opts ...testutil.GetDBOption) (http.Handler, *depe
 	meta := auth.NewDBMetadataManager("dev", conn)
 	retentionService := retention.NewService(conn)
-	migrator := db.NewDatabaseMigrator(dbparams.Database{URI: handlerDatabaseURI})
+	migrator := db.NewDatabaseMigrator(dbparams.Database{ConnectionString: handlerDatabaseURI})
```
```suggestion
	migrator := db.NewDatabaseMigrator(dbparams.Database{Driver: db.DatabaseDriver, ConnectionString: handlerDatabaseURI})
```
(and throughout)

arielshaqed commented on api/handler_test.go:
```diff
@@ -86,7 +86,7 @@ func getHandler(t *testing.T, opts ...testutil.GetDBOption) (http.Handler, *depe
 	meta := auth.NewDBMetadataManager("dev", conn)
 	retentionService := retention.NewService(conn)
-	migrator := db.NewDatabaseMigrator(dbparams.Database{URI: handlerDatabaseURI})
+	migrator := db.NewDatabaseMigrator(dbparams.Database{ConnectionString: handlerDatabaseURI})
```
I remain confused about whether this is intended to be used as a URI or as a connection string.  I *think* it is a connection string, so we should call it that throughout the code and in YaML config etc.

arielshaqed replied:
This could also be in a later PR, of course, if we want to get this one of our plates.

nopcoder replied:
The configuration and code are now aligned - connection_string (yaml) and ConnectionString (go)

arielshaqed reviewed (approved):
Thanks!  LGTM.
